server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/operationhandler/EcoreDeleteOperationHandler.java

```diff
@@ -56,7 +60,27 @@ public void executeOperation(final DeleteOperation operation, final GModelState
 								"Could not execute delete operation on EAttribute: " + element.toString());
 					}
 				}
-
+			}, () -> {
+				if (EcoreEdgeUtil.isGeneralizationEdge(elementId)) {
+					Optional<EObject> baseClass = modelState.getIndex()
+							.getSemantic(EcoreEdgeUtil.getBaseClassId(elementId));
+					baseClass.ifPresent(base -> {
+						Optional<EObject> superClass = modelState.getIndex()
+								.getSemantic(EcoreEdgeUtil.getSuperClassId(elementId));
+						superClass.ifPresent(superType -> {
+							if (base instanceof EClass && superType instanceof EClass
+									&& ((EClass) base).getESuperTypes().contains(superType)) {
+								EGenericType genericSuperType = ((EClass) base).getEGenericSuperTypes().stream()
```

Comment:
EGenericType is the equivalent to Java's `Class<T>`. I don't think it's relevant here (as we currently don't support EGenericTypes in the Ecore editor). You can use ((EClass)base).getESuperTypes() everywhere (It's a list of EClass)